api/api/openapi.py

```diff
@@ -1,38 +1,68 @@
-from copy import deepcopy
-from functools import lru_cache
-
-import jsonref
-from drf_yasg.openapi import Schema
-from flag_engine.environments.models import EnvironmentModel
-
-SKIP_PROPERTIES = [
-    "amplitude_config",
-    "dynatrace_config",
-    "heap_config",
-    "mixpanel_config",
-    "rudderstack_config",
-    "segment_config",
-    "webhook_config",
-]
-SKIP_DEFINITIONS = ["IntegrationModel", "WebhookModel"]
-
-
-@lru_cache()
-def get_environment_document_response() -> Schema:
-    model_json_schema = EnvironmentModel.model_json_schema(mode="serialization")
-
-    # Restrict segment rule recursion to two levels.
-    segment_rule_schema = deepcopy(model_json_schema["$defs"]["SegmentRuleModel"])
-    del segment_rule_schema["properties"]["rules"]
-    model_json_schema["$defs"]["SegmentRuleInnerModel"] = segment_rule_schema
-    model_json_schema["$defs"]["SegmentRuleModel"]["properties"]["rules"]["items"][
-        "$ref"
-    ] = "#/$defs/SegmentRuleInnerModel"
-
-    # Remove integrations.
-    for prop in SKIP_PROPERTIES:
-        del model_json_schema["properties"][prop]
-    for definition in SKIP_DEFINITIONS:
-        del model_json_schema["$defs"][definition]
-
-    return Schema(**jsonref.replace_refs(model_json_schema))
+from typing import Any
+
+from drf_yasg.inspectors import SwaggerAutoSchema
+from drf_yasg.openapi import SCHEMA_DEFINITIONS, Response, Schema
+from pydantic import BaseModel
+from pydantic.json_schema import GenerateJsonSchema, JsonSchemaValue
+from pydantic_core import core_schema
+
+
+class _GenerateJsonSchema(GenerateJsonSchema):
+    def nullable_schema(self, schema: core_schema.NullableSchema) -> JsonSchemaValue:
+        """Generates an OpenAPI 2.0-compatible JSON schema that matches a schema that allows null values.
+
+        Args:
+            schema: The core schema.
+
+        Returns:
+            The generated JSON schema.
+        """
+        anyof_schema_value = super().nullable_schema(schema)
+        elem = next(
+            any_of
+            for any_of in anyof_schema_value["anyOf"]
+            if any_of.get("type") != "null"
+        )
+        if type := elem.get("type"):
+            return {"type": type, "x-nullable": True}
+        # Assuming a reference here (which we can not annotate)
+        return elem
+
+
+class PydanticResponseCapableSwaggerAutoSchema(SwaggerAutoSchema):
+    def get_response_schemas(
+        self,
+        response_serializers: dict[str | int, Any],
+    ) -> dict[str, Response]:
+        result = {}
+
+        for status_code in list(response_serializers):
```

Comment:
I'm not sure why we elected not to iterate over `response.serializers.items()` here?